### PR TITLE
PIC-3173 add assigned to filter

### PR DESCRIPTION
**Pull request by rschallapalli**

### CODE REVIEW

File: src/main/java/uk/gov/justice/probation/courtcaseservice/jpa/entity/HearingEntity.java [hunk 1]
```diff
@@ -19,21 +19,9 @@
...
-import javax.persistence.GeneratedValue;
-import javax.persistence.GenerationType;
-import javax.persistence.Id;
-import javax.persistence.JoinColumn;
-import javax.persistence.ManyToOne;
-import javax.persistence.OneToMany;
-import javax.persistence.OneToOne;
-import javax.persistence.OrderBy;
-import javax.persistence.Table;
+import javax.persistence.*;
```
Comment (paddybasi) [inline src/main/java/uk/gov/justice/probation/courtcaseservice/jpa/entity/HearingEntity.java:22]: I would've assumed it's best to use named imports, ```*``` means importing everything?

Comment (rschallapalli) [reply src/main/java/uk/gov/justice/probation/courtcaseservice/jpa/entity/HearingEntity.java:22]: Had to use a native query given the complexity is not supported by ORM framework.

File: src/main/kotlin/uk/gov/justice/probation/courtcaseservice/jpa/repository/HearingOutcomeRepositoryCustom.kt [hunk 2]
```diff
@@ -14,6 +15,74 @@ import javax.persistence.criteria.Predicate
 @Repository
 class HearingOutcomeRepositoryCustom(private val entityManager: EntityManager) {
 
+    fun findByCourtCodeAndHearingOutcome2(
+        courtCode: String,
+        hearingOutcomeSearchRequest: HearingOutcomeSearchRequest
+    ): List<Pair<HearingEntity, LocalDate>> {
+
+        var filterBuilder = StringBuilder();
```
[the remark this answers is not in the document]
Comment (rschallapalli) [reply src/main/kotlin/uk/gov/justice/probation/courtcaseservice/jpa/repository/HearingOutcomeRepositoryCustom.kt:23]: Had to use a native query given the complexity is not supported by ORM framework.

File: src/main/java/uk/gov/justice/probation/courtcaseservice/jpa/entity/HearingEntity.java (hunk 1 above)
Comment (rschallapalli) [reply src/main/java/uk/gov/justice/probation/courtcaseservice/jpa/entity/HearingEntity.java:22]: Yes, IDE did this thing. will revert

**Review by paddybasi — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/probation/courtcaseservice/jpa/repository/HearingOutcomeRepositoryCustom.kt [hunk 3]
```diff
@@ -14,6 +15,74 @@ import javax.persistence.criteria.Predicate
...
+        }
+
+        hearingOutcomeSearchRequest.outcomeType?.let { outcomeTypes ->
+            if(outcomeTypes.isNotEmpty()) {
+                jpaQuery.setParameter("outcomeTypes", hearingOutcomeSearchRequest.outcomeType.map { it.name })
+            }
+        }
+
+        var result = jpaQuery.resultList.map {it as Array<Any>}.map {Pair(it[0] as HearingEntity, it[1] as LocalDate)}
+        return result;
```
Comment (vb-moj) [inline src/main/kotlin/uk/gov/justice/probation/courtcaseservice/jpa/repository/HearingOutcomeRepositoryCustom.kt:83]: Could be just  `return jpaQuery.resultList.map {it as Array<Any>}.map {Pair(it[0] as HearingEntity, it[1] as LocalDate)}`?

File: src/main/kotlin/uk/gov/justice/probation/courtcaseservice/jpa/repository/HearingOutcomeRepositoryCustom.kt [hunk 4]
```diff
@@ -41,8 +117,9 @@ class HearingOutcomeRepositoryCustom(private val entityManager: EntityManager) {
         criteriaQuery.where(*restrictions.toTypedArray())
 
         processSorting(hearingOutcomeSearchRequest, hearingDay, criteriaQuery, criteriaBuilder)
-        return entityManager.createQuery(criteriaQuery)
+        val resultList = entityManager.createQuery(criteriaQuery)
             .resultList
+        return resultList
```
Comment (vb-moj) [inline src/main/kotlin/uk/gov/justice/probation/courtcaseservice/jpa/repository/HearingOutcomeRepositoryCustom.kt:122]: same as the above

File: src/main/kotlin/uk/gov/justice/probation/courtcaseservice/service/CaseWorkflowService.kt [hunk 5]
```diff
@@ -42,6 +42,6 @@ class CaseWorkflowService(val hearingRepository: HearingRepository, val courtRep
                     courtCode
                 )
             }
-        return hearingOutcomeRepositoryCustom.findByCourtCodeAndHearingOutcome(courtCode, hearingOutcomeSearchRequest).flatMap { HearingOutcomeResponse.of(it) }
+        return hearingOutcomeRepositoryCustom.findByCourtCodeAndHearingOutcome2(courtCode, hearingOutcomeSearchRequest).flatMap { HearingOutcomeResponse.of(it.first, it.second) }
```
Comment (vb-moj) [inline src/main/kotlin/uk/gov/justice/probation/courtcaseservice/service/CaseWorkflowService.kt:45]: Is this renaming intentional?

File: src/main/kotlin/uk/gov/justice/probation/courtcaseservice/jpa/repository/HearingOutcomeRepositoryCustom.kt [hunk 6]
```diff
@@ -14,6 +15,74 @@ import javax.persistence.criteria.Predicate
 @Repository
 class HearingOutcomeRepositoryCustom(private val entityManager: EntityManager) {
 
+    fun findByCourtCodeAndHearingOutcome2(
+        courtCode: String,
+        hearingOutcomeSearchRequest: HearingOutcomeSearchRequest
+    ): List<Pair<HearingEntity, LocalDate>> {
```
[the remark this answers is not in the document]
Comment (rschallapalli) [reply src/main/kotlin/uk/gov/justice/probation/courtcaseservice/jpa/repository/HearingOutcomeRepositoryCustom.kt:21]: I guess this is the best way or I can do the DTO mapping here which again breaks some principle. with tupples being typed I don't see an issue

File: src/main/kotlin/uk/gov/justice/probation/courtcaseservice/jpa/repository/HearingOutcomeRepositoryCustom.kt [hunk 7]
```diff
@@ -14,6 +15,74 @@ import javax.persistence.criteria.Predicate
 @Repository
 class HearingOutcomeRepositoryCustom(private val entityManager: EntityManager) {
 
+    fun findByCourtCodeAndHearingOutcome2(
```
[the remark this answers is not in the document]
Comment (rschallapalli) [reply src/main/kotlin/uk/gov/justice/probation/courtcaseservice/jpa/repository/HearingOutcomeRepositoryCustom.kt:18]: good spot

File: src/main/kotlin/uk/gov/justice/probation/courtcaseservice/jpa/repository/HearingOutcomeRepositoryCustom.kt (hunk 4 above)
Comment (rschallapalli) [reply src/main/kotlin/uk/gov/justice/probation/courtcaseservice/jpa/repository/HearingOutcomeRepositoryCustom.kt:122]: will do

File: src/main/kotlin/uk/gov/justice/probation/courtcaseservice/service/CaseWorkflowService.kt (hunk 5 above)
Comment (rschallapalli) [reply src/main/kotlin/uk/gov/justice/probation/courtcaseservice/service/CaseWorkflowService.kt:45]: No, I left the code untidy and raised PR carelessly as soon as test passed. Apologies

**Pull request merged**